### DISCUSSION TITLE
"HmIP-eTRV-2 I9F" erlauben keine Direktverknüpfungen

### DISCUSSION THREAD

**Discussion by t-offline** [OpenCCU/OpenCCU#2310] [🇩🇪 Fragen&Antworten]
Hallo, ich habe ein Problem mit der Erzeugung von "HmIP-Heating control".
Meine beiden HmIP-eTRV-2 werden nicht eingebunden.
Es werden also keine "Direct device connections" zwischen den beiden HmIP-eTRV-2 erzeugt.

In RaspberryMatic-3.67.10.20230225-rpi3.zip geht es noch (downgrade gemacht).

In RaspberryMatic-3.69.6.20230407-rpi3.zip geht es dann nicht mehr,
und in RaspberryMatic-3.69.7.20230506-rpi3.zip auch nicht.

Was mache ich falsch?

P.S.
"Direct device connections" manuell geht auch nicht, seit Version 3.69.6.20230407 meldet es nur:
"There are no channels available which can be connected."
Ich habe alle Tests mit Reset auf Werkseinstellung gemacht.

**Comment by jens-maus** [comment]
Tut mir leid, aber das sind wirklich zu wenig Detailinformationen um hier sinnvoll helfen zu können. Bitte zeig am besten ein Screenshot deines Problems was du hier genau meinst mit "werden nicht eingebunden"?!?.

**Reply by t-offline** [reply]
![photo_2023-05-08_11-36-58](https://user-images.githubusercontent.com/4408267/236790622-22b8e1a7-72bc-4be1-a671-85ac30088fb3.jpg)

Hallo,

es geht darum:

Meine "eQ-2 14028A0E HmIP-eTRV-2 I9F" lassen sich nicht als 
"Direct device connections directly connect two or more devices with each other" anwählen. Somit geht auch keine "HmIP-Heating control" group.

Das Problem ist in der Version 3.67.10.20230225 nicht vorhanden.
In den beiden neueren Versionen leider schon.

Dort wird bei "New connetion" und Auswahl eines ersten Partners "HmIP-eTRV-2 I9F 00395F29A6428E:1" kein zweiter Partner angeboten.
Da ich jetzt meine beiden RaspberryMatic Systeme auf Version 3.67.10.20230225 downgegradet habe, kann ich leider keinen screenshot vom Fehler liefern.

Der screenshot ist mit Version 3.67.10.20230225 gemacht, wo es noch geht.

**Reply by jens-maus** [reply]
Versteh den Screenshot und deine Beschreibung leider nicht ganz. Was genau willst du denn tun? Welches Gerät willst du mit welchem nun via Direktverknüpfung verknüpfen und über welchen Weg? D.h. über "Settings -> Direct connections" oder versuchst du Gerät in einer Heating Group hinzuzufügen? Dann zeig bitte mal ein Screenshot bzw. Screencast mit der 3.67.10.20230225 version wo man sieht wie du dort die Verbindung anlegst und das Gerät auftaucht das du verknüpfen willst und dann eben mit der aktuellen 3.69.7.20230506 wo zu sehen ist das das Gerät fehlt in der Auflistung.

**Comment by t-offline** [comment]
![photo_2023-05-08_12-15-55](https://user-images.githubusercontent.com/4408267/236800499-115a93ba-b7b6-46e8-8096-53087542c509.jpg)
![photo_2023-05-08_12-16-00](https://user-images.githubusercontent.com/4408267/236800504-f63b6d8f-b8ba-4df2-9271-0d3fb8ab23c1.jpg)
![photo_2023-05-08_12-16-09](https://user-images.githubusercontent.com/4408267/236800510-3748062f-36e1-4161-bd8f-44b2124e8a0d.jpg)
![photo_2023-05-08_12-16-16](https://user-images.githubusercontent.com/4408267/236800512-63b98f14-b920-4695-958f-03cf2f43668c.jpg)
![photo_2023-05-08_12-16-20](https://user-images.githubusercontent.com/4408267/236800514-091c0214-e3c5-42f6-89d2-bd3f25a69823.jpg)
![photo_2023-05-08_12-16-25](https://user-images.githubusercontent.com/4408267/236800518-92d0123f-c5e3-4474-a1f9-d1f0a648ed2b.jpg)
![photo_2023-05-08_12-16-29](https://user-images.githubusercontent.com/4408267/236800519-a17c4218-23ed-4fbb-93a1-15699fadbeca.jpg)
![photo_2023-05-08_12-16-36](https://user-images.githubusercontent.com/4408267/236800520-656a3eb6-553f-4aac-8c0b-b68e67caf77f.jpg)
![photo_2023-05-08_12-16-41](https://user-images.githubusercontent.com/4408267/236800521-ab5e340b-6fd5-4b29-b70b-e3d42d13d542.jpg)
![photo_2023-05-08_12-16-45](https://user-images.githubusercontent.com/4408267/236800523-4b89c1ee-fc3a-4ef2-a94c-873c1d5279f2.jpg)
![photo_2023-05-08_12-16-52](https://user-images.githubusercontent.com/4408267/236800524-9e36ccec-bfc5-4d9b-9c84-4002a11e7c59.jpg)

Hallo,

jetzt versuche ich das Problem zu reduzieren.

Also erstmal keine heating group.

Nur neue Direktverknüpfung zwischen zwei HmIP-eTRV-2 I9F.

Da ich, wie gesagt z.Z. nur ein Testsystem auf Version 3.67.10.20230225 (wo es noch geht) habe, zeige ich was hier noch geht, aber in den höheren Versionen nicht mehr.
Dort kommt immer nur "There are no channels available which can be connected." nach der Auswahl des ersten HmIP-eTRV-2 I9F.

**Reply by jens-maus** [reply]
Ok danke. Ich hab aber leider solch ein `HmIP-eTRV-2 I9F` nicht zum testen und kann das daher hier nicht nachstellen. Kann gut sein das das entweder ggf. nur ein WebUI Anzeigeproblem ist und das Gerät blos nicht in der Auflistung auftaucht seit neueren Versionen oder es liegt doch am closed source Teil `HMIPServer` der auch die Gruppen managed und daher irgendwie mit diesen neuen I9F Typen des HmIP-eTRV-2 nicht klarkommt. Leider hat sich zu meinem totalen unverständnis eQ3 bei der Entwicklung dieses neuen I9F Types dazu hinreissen lassen ein Leerzeichen in die Typenbeschreibung mit aufzunehmen. Würde mich nicht wundern wenn Ihnen das jetzt hier+da auf die Füße fällt.... Was du noch probieren könntest ist es mit der Originalen CCU3 3.69.7 Firmware nachzuspielen um zu sehen ob es da prinzipiell geht, denn dann wäre das ein Hinweis darauf das es in der Tat an RaspberryMatic liegt (was ich allerdings nicht glaube)...

**Comment by t-offline** [comment]
Hallo,
ich habe eine sd-card geopfert und an meinem pi3 folgendes installiert:

> pi@raspberrypi:~ $ sudo pivccu-info
piVCCU version: 3.69.7-80
Kernel modules: Available
Raw UART dev:   Available
Rasp.Pi UART:   Assigned to GPIO pins
HMRF Hardware:  HM-MOD-RPI-PCB
 Connected via: GPIO@3f201000.serial (/dev/raw-uart)
 Board serial:  SEQ1773241
 Radio MAC:     0x752005
HMIP Hardware:  HM-MOD-RPI-PCB
 Connected via: GPIO@3f201000.serial (/dev/raw-uart)
 SGTIN:         3014F711A061A7DD89A476B9
 Radio MAC:     0xB6C5EA
State:          RUNNING
PID:            2372
IP:             192.168.244.51
CPU use:        48.76 seconds
Link:           vethpivccu
 TX bytes:      6.57 MiB
 RX bytes:      2.02 MiB
 Total bytes:   8.59 MiB


Gleiches Problem!

![photo_2023-05-08_15-21-40](https://user-images.githubusercontent.com/4408267/236836292-c49cac4b-feb0-40f5-8604-1fcf292c5adc.jpg)
![photo_2023-05-08_15-21-03](https://user-images.githubusercontent.com/4408267/236836298-e69558d5-7868-460f-b786-8f3b07fb5856.jpg)
![photo_2023-05-08_15-21-06](https://user-images.githubusercontent.com/4408267/236836301-0f3fbe18-2661-4769-83ee-b9dfa26d35e0.jpg)
![photo_2023-05-08_15-21-10](https://user-images.githubusercontent.com/4408267/236836303-37d91217-76fd-40a4-a817-952e59f3d00b.jpg)
![photo_2023-05-08_15-21-13](https://user-images.githubusercontent.com/4408267/236836308-b5d2cb6e-e5c3-4076-b594-124cbe28fe10.jpg)
![photo_2023-05-08_15-21-17](https://user-images.githubusercontent.com/4408267/236836309-e3b97b63-47f9-4f9e-ba26-b63eb31d1255.jpg)
![photo_2023-05-08_15-21-21](https://user-images.githubusercontent.com/4408267/236836310-dd13bb3c-0e36-4d51-a108-5f94f10b8879.jpg)
![photo_2023-05-08_15-21-25](https://user-images.githubusercontent.com/4408267/236836312-803e2f77-0a79-4727-92ae-8833c78225bc.jpg)
![photo_2023-05-08_15-21-28](https://user-images.githubusercontent.com/4408267/236836315-ad6035d3-b227-41be-bacc-ccebec54ae2f.jpg)
![photo_2023-05-08_15-21-34](https://user-images.githubusercontent.com/4408267/236836319-2b03800e-30c0-47e4-9c69-4a925dd6e957.jpg)
![photo_2023-05-08_15-21-38](https://user-images.githubusercontent.com/4408267/236836322-a975f48b-0f35-41a4-addb-f78b61b8b3c8.jpg)


Was kann ich noch zur Lösung der Problems beitragen?

P.S.
Danke für Deine Mühe

**Reply by jens-maus** [reply]
Dann scheint das in der Tat ein Problem in neueren Firmwareversionen zu sein das der `HmIP-eTRV-2 I9F` nicht auswählbar ist bei den Direktverknüpfungen. Vielleicht haben ja @jp112sdl oder @Baxxy13 noch eine Idee bzw. Testgeräte um das zu testen oder analysieren. Meine persönliche Idee ist, das es sicherlich an diesem unleidlichen Leerzeichen in der Typenbezeichnung liegt warum die WebUI an einer Stelle durcheinander kommt und dann diese nicht zur Auswahl präsentiert....

**Reply by jp112sdl** [reply]
Welcher Gerätetyp als Empfänger-Partner sollte bei Kanal 1 als Sender auswählbar sein?

Und warum steht das HM-MOD-UART als "Unknown Device" in der Geräteliste?
Das passiert eigentlich nur, wenn man manuell in der `crRFD.conf` den Parameter
`Adapter.Local.Device.Enabled=true` setzt.
Beim HM-MOD-RPI-PCB wird diese Option durch das Init-Skript eigentlich auf `false` gesetzt.
Wurde in der Config evtl noch mehr rumgespielt?
Möglicherweise führt das auch zu unerwünschtem Verhalten innerhalb des HMIPServers.

**Comment by Baxxy13** [comment]
Laut Handbuch werden mehrere eTRV's über Kanal:1 und Kanal:2 "kreuzverknüpft".
Also:
eTRV "A" CH:1 <--> eTRV "B" CH:2 & eTRV "A" CH:2 <--> eTRV "B" CH:1

Laut Device Doku hat:
CH:1 als "Link Source Type": "CLIMATE_CONTROL"
CH:2 als "Link Target Type": "CLIMATE_CONTROL_WGD_WTH

Selber testen kann ich mangels Geräte nicht.

Wie sieht's denn mit Verknüpfungen zwischen eTRV-2 und eTRV-2 I9F aus, geht da nur "die Hälfte"?

Versuchen könnte man die DV's mit dem Homematic-Manager (so man ihn zum laufen bekommt) anzulegen. 
Der kann (nicht immer) die Restriktionen der WebUI umgehen.

Oder direkt auf der ssh-Konsole versuchen:
`echo 'load tclrpc.so; puts [xmlrpc http://127.0.0.1:32010 addLink [list string 00_SN_etrv_A:1] [list string 00_SN_etrv_B:2]]' | tclsh`

**Comment by t-offline** [comment]
Hallo,

ich habe etwas mit "Homematic Manager Version 2.7.1" gespielt.
Aus Windows 10 ging es nicht.
Aber auf meinem geliebten Windows 7 ohne Probleme.

Als CCU habe ich einen rp3 mit "piVCCU3" und darin im LXC dann die CCU3 am laufen.

Hier meine erfolglosen Versuche eine Direktverknüpfung zwischen zwei "HmIP-eTRV-2 I9F" herzustellen.
Geht nicht!
Einen "HmIP-eTRV-2", also ohne " I9F" habe ich nicht.


Hinweis: Das mit dem HM-MOD-UART als "Unknown Device" halte ich für "Kosmetik", da habe ich nicht " rumgespielt"

> root@raspberrypi:~# pivccu-info
piVCCU version: 3.69.7-80
Kernel modules: Available
Raw UART dev:   Available
Rasp.Pi UART:   Assigned to GPIO pins
HMRF Hardware:  HM-MOD-RPI-PCB
 Connected via: GPIO@3f201000.serial (/dev/raw-uart)
 Board serial:  SEQ1773241
 Radio MAC:     0x752005
HMIP Hardware:  HM-MOD-RPI-PCB
 Connected via: GPIO@3f201000.serial (/dev/raw-uart)
 SGTIN:         3014F711A061A7DD89A476B9
 Radio MAC:     0xBBC98D
State:          RUNNING
PID:            919
IP:             192.168.244.51
IP:             2001:9e8:a636:d600:a801:b5ff:fe12:b2b7
IP:             fdfb:244::a801:b5ff:fe12:b2b7
CPU use:        113.02 seconds
Link:           vethpivccu
 TX bytes:      13.58 MiB
 RX bytes:      8.72 MiB
 Total bytes:   22.30 MiB


![photo_2023-05-09_09-50-36](https://user-images.githubusercontent.com/4408267/237031447-7a2de38e-feb6-48cf-aad2-c2e0aa529a36.jpg)
![photo_2023-05-09_09-50-46](https://user-images.githubusercontent.com/4408267/237031451-4868184e-cfc7-4639-bdf6-cdfab9331b17.jpg)
![photo_2023-05-09_09-50-51](https://user-images.githubusercontent.com/4408267/237031454-0684502b-88c1-4f9e-9a58-dfc565d78cb0.jpg)
![photo_2023-05-09_09-50-55](https://user-images.githubusercontent.com/4408267/237031456-ffddd2e7-cafa-470b-a759-b8fbe79e2cee.jpg)
![photo_2023-05-09_09-51-00](https://user-images.githubusercontent.com/4408267/237031460-b053efe7-8023-41b3-9a3a-33bce745f8db.jpg)
![photo_2023-05-09_09-51-03](https://user-images.githubusercontent.com/4408267/237031463-4d37e6d4-6177-4bde-b773-33d6a310aee3.jpg)
![photo_2023-05-09_09-51-08](https://user-images.githubusercontent.com/4408267/237031465-2ce6e832-b0c4-479e-9fa7-68aa900ad019.jpg)
![photo_2023-05-09_09-51-11](https://user-images.githubusercontent.com/4408267/237031466-7124f016-f31d-4e06-9372-c934ad70b5be.jpg)
![photo_2023-05-09_09-51-15](https://user-images.githubusercontent.com/4408267/237031468-66d504c6-f6a5-44cf-a1ee-efbc23f74bca.jpg)
![photo_2023-05-09_09-51-18](https://user-images.githubusercontent.com/4408267/237031470-58eecbae-0b11-4df4-8d63-9f36c4ecff3c.jpg)
![photo_2023-05-09_09-51-22](https://user-images.githubusercontent.com/4408267/237031473-ba8fa988-bf74-4df3-94a1-c8be2f8a67eb.jpg)
![photo_2023-05-09_09-51-26](https://user-images.githubusercontent.com/4408267/237031475-75c788c8-4fe8-4c54-8286-ba1c7ce211fe.jpg)
![photo_2023-05-09_09-51-30](https://user-images.githubusercontent.com/4408267/237031444-ebfdfaaf-28b6-47f4-a946-b75630b93bc2.jpg)

**Reply by Baxxy13** [reply]
Hmm, schade. 
Dann wir das ssh Kommando auch nichts bringen denn es macht nix anderes wie der Homematic Manager. 
Probieren kannst du es natürlich trotzdem.

Sieht mir dann jetzt auch nach einem Problem im HmIP-Server aus. Der scheint das nicht "zu erlauben", vielleicht weil da irgendwas mit der Device Description nicht stimmt.

Thema für eQ-3

**Reply by t-offline** [reply]
Danke, sehe ich auch so. Habe bei eQ-3 schon ein Ticket aufgemacht, mit Verweis hierher. Dann warten wir mal ab....

**Reply by jens-maus** [reply]
Da wird sicher prompt die Antwort kommen das sie für Systeme die keine CCU3 Firmware nutzen kein Support leisten. D.h. wenn dann musst du das (wenn du das schon selbst direkt an eQ3 melden willst) mit einer barebone CCU3 samt originaler CCU3 Firmware reproduzieren und denen dann so melden...

Ich hab es aber parallel schon einmal an meinen direkten eQ3 Kontakt gemeldet.. mal sehen ob da was dann passiert.

**Reply by jp112sdl** [reply]
> Sieht mir dann jetzt auch nach einem Problem im HmIP-Server aus. Der scheint das nicht "zu erlauben", vielleicht weil da irgendwas mit der Device Description nicht stimmt.

Da wurde aber nichts verändert seit der 3.67.10
HmIP-eTRV
HmIP-eTRV-2
HmIP-eTRV-2 I9F
und -UK
verwenden ein und dieselbe XML

**Comment by Baxxy13** [comment]
Ohne RPI-RF-MOD ist es unmöglich eine CCU3 zu erzeugen. Mit allen alternativen Funkmodulen startet die nicht sauber. 
Wenn ich es schaffe poste ich später einen "Hilferuf" im Forum, vielleicht findet sich jemand der das mit Original CCU3 und entsprechenden Geräten reproduzieren kann.

**Reply by jp112sdl** [reply]
> Hinweis: Das mit dem HM-MOD-UART als "Unknown Device" halte ich für "Kosmetik", da habe ich nicht " rumgespielt"

@t-offline Kam das von allein? Dann ist es ein Bug in RaspberryMatic.
Bei meinem RPi3B mit HM-MOD-RPI-PCB ist dieses Gerät nicht vorhanden.

**Reply by jp112sdl** [reply]
Ich hatte mir diese "Kosmetik" selbst mal hingefummelt und kann bestätigen, was Baxxy schreibt: Es starten dann nicht alle Worker-Prozesse im HMIPServer sauber.

Also so lange das nicht "behoben" ist, bin ich hier auch raus beim weiteren Suchen.

Und da kann ich dann auch eQ-3 verstehen, wenn sie den Support ablehneen.

**Reply by jens-maus** [reply]
> > Hinweis: Das mit dem HM-MOD-UART als "Unknown Device" halte ich für "Kosmetik", da habe ich nicht " rumgespielt"
> 
> @t-offline Kam das von allein? Dann ist es ein Bug in RaspberryMatic. Bei meinem RPi3B mit HM-MOD-RPI-PCB ist dieses Gerät nicht vorhanden.

Er hatte sich doch ein piVCCU installiert zum testen mit "originaler CCU3 Firmware". Und da taucht das anscheinend immer auf...

**Reply by Baxxy13** [reply]
Das `true` ist ja bei CCU3 neuerdings Default. Sie kennt ja auch kein anderes Funkmodul. Und wenn Alex da bei piVCCU nicht irgendwie eingreift dann bleibt es halt bei `true`. 
Ob das jetzt tatsächlich problematisch ist weiß ich aber auch nicht.

**Reply by jp112sdl** [reply]
Na dann mal der Reihe nach.
Die Meldung "invalid access point, device or channel" kommt, wenn einer der Fehler
- `WRONG_ACCESS_POINT`
- `UNKNOWN_ACCESS_POINT`
- `UNKNOWN_DEVICE`
- `INVALID_CHANNEL`

auftritt.

Die ersten beiden Punkte haben mit dem eigenen Funkmodul zu tun.
Die anderen beiden mit dem Gerät.

`UNKNOWN_DEVICE` schließen wir mal aus, da die Seriennummern direkt aus dem System gelesen werden und das auch nur erfolgen kann, wenn die Geräte angelernt sind.

Bliebe noch `INVALID_CHANNEL`
Das kommt, wenn
- die `LINK_ROLE` von Gerät A == `LINK_ROLE` von Gerät B ist (also wenn man `SENDER` mit `SENDER` oder `RECEIVER` mit `RECEIVER` verknüpfen wollen würde)
- der Typ der `LINK_ROLE` von Gerät A nicht dem Typ der `LINK_ROLE` von Gerät B entspricht

Im Fall des eTRV ist
- `Kanal 1` vom Typ `HEATING_CLIMATECONTROL_TRANSCEIVER` mit der `LINK_ROLE`=`SENDER` und Typ `CLIMATE_CONTROL`
- `Kanal 2` vom Typ `HEATING_CLIMATECONTROL_RECEIVER` mit der `LINK_ROLE`=`RECEIVER` und Typ `CLIMATE_CONTROL`

Damit _dürfte_ die Exception `INVALID_CHANNEL` nicht geworfen werden, wenn man 2 Geräte verknüpfen will, unter Verwendung von Kanal 1 + 2.
Bzw. wenn es damit zu tun hat, müsste der Fehler auch beim normalen `eTRV` oder `eTRV-2` auftreten.

Für mich bleiben dann nur die Sachen mit dem Funkmodul (`WRONG_ACCESS_POINT`, `UNKNOWN_ACCESS_POINT`)...

Mehr fällt mir dann aber auch nicht zum Debuggen ein.

Bin gespannt, was eQ-3 meint und woran es letztendlich liegt

**Reply by jp112sdl** [reply]
Ok, ich lag daneben mit dem Funkmodul.

Es zeichnete [hier im Forum](https://homematic-forum.de/forum/viewtopic.php?f=60&t=78885&p=766984&#p766953) schon ab, dass da intern irgendwas mit den Link-Rollen daneben läuft.

Der Fehler steckt in der Verarbeitung der Channel Description Files im Zusammenspiel mit der neu hinzu gekommenen `channel_type_heating_climatecontrol_receiver_scthd.xml`

Die `channel_type_heating_climatecontrol_receiver_trv.xml` und auch die `channel_type_heating_climatecontrol_receiver_scthd.xml` definieren gleichermaßen den `HEATING_CLIMATECONTROL_RECEIVER` mit der Rolle `RECEIVER`.

- `channel_type_heating_climatecontrol_receiver_trv.xml` mit Typ `CLIMATE_CONTROL`
- `channel_type_heating_climatecontrol_receiver_scthd.xml` mit Typ `CLIMATE_CONTROL_WGD_WTH`

Die `channel_type_heating_climatecontrol_receiver_scthd.xml` kam erst jetzt mit der 3.69 hinzu und bei der internen Verarbeitung im HMIPServer gewinnt der letzte `HEATING_CLIMATECONTROL_RECEIVER`... in dem Fall mit dem Link-Typ `CLIMATE_CONTROL_WGD_WTH`.

Danke an @Baxxy13 fürs Testen

**Comment by Baxxy13** [comment]
So, der ["Hilferuf"](https://homematic-forum.de/forum/viewtopic.php?f=60&t=78885#p766447) ist raus.

Mal schauen ob sich jemand findet.

**Reply by t-offline** [reply]
Danke an alle fürs Testen.
Servus Christian

**Comment by Baxxy13** [comment]
Ich lehne mich mal aus dem Fenster und sage in der nächsten CCU3-Firmware ist das Problem gefixt.
(zumindest im RM-Nightly 3.71.6.20230701 sieht das danach aus)

**Reply by jens-maus** [reply]
So sollte es sein, ja ;)
